### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by monicah79**
- Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).
- Set up associations between models.
Remember that author_id column in posts table should be the foreign_key for the users table.
-My models  includes some additional methods.

Users
A method that returns the 3 most recent posts for a given user.
Posts
A method that updates the posts counter for a user.
A method which returns the 5 most recent comments for a given post.
Comments
A method that updates the comments counter for a post.
Likes
A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  validates :name, :photo, :bio, presence: true
+  validates :posts_counter, numericality: { greater_than_or_equal_to: 0 }
+  attr_accessor :photo, :bio, :email
```
Comment (codepantha) [inline app/models/user.rb]: - [ ] Well done for creating the association between user and posts 👏. Please also do the same for:
1. user and comments
2. user and likes

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  private
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_count
+    self.posts_count = posts.count
+  end
```
Comment (codepantha) [inline app/models/user.rb]: - [ ] The method to update the user's posts count is implemented in the `Post` model and not in the `User` model according to requirements. Kindly remove this code 👍

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class Comment < ApplicationRecord
+  belongs_to :post, counter_cache: true
+  belongs_to :author, class_name: 'User'
+  belongs_to :user
+
+  def update_comments_counter
+    post.comments_counter += 1
+    post.save
+  end
```
Comment (codepantha) [inline app/models/comment.rb:15]: - [ ] Just like you did in the `Post` model, please use `after_save :update_comments_counter`

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class Like < ApplicationRecord
+  belongs_to :post, counter_cache: true
+  belongs_to :user
+
+  def update_likes_counter
+    post.update(likes_counter: post.likes_counter + 1)
+  end
```
Comment (codepantha) [inline app/models/like.rb:11]: - [ ] Just like you did in the Post model, please use `after_save :update_likes_counter`

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @monicah79 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Some associations between models have been set up 📌 
✔️ Linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by addod19 — APPROVED** [review]
### Hi @monicah79, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Awesome job fixing all the comments from the previous review :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**